### PR TITLE
Improve PUT idea endpoints

### PR DESCRIPTION
**Pull request by ottaviano**

### CODE REVIEW

File: features/api/ideas_workshop_ideas.feature [hunk 1]
```diff
@@ -823,17 +823,12 @@ Feature:
   Scenario: As a logged-in user I can not publish an idea that has another status than PENDING at the moment of execution
     Given I am logged as "jacques.picard@en-marche.fr"
     When I send a "PUT" request to "/api/ideas/4/publish"
-    Then the response status code should be 403
+    Then the response status code should be 400
 
   Scenario: As a logged-in user I get errors when I publish my idea with few data
```
Comment (illusionOfParadise) [inline features/api/ideas_workshop_ideas.feature]: Please, change also this scenario description.

File: features/api/ideas_workshop_ideas.feature [hunk 2]
```diff
@@ -823,17 +823,12 @@ Feature:
   Scenario: As a logged-in user I can not publish an idea that has another status than PENDING at the moment of execution
```
Comment (illusionOfParadise) [inline features/api/ideas_workshop_ideas.feature]: Please, change this scenario description: only idea in the DRAFT status can be published.

File: features/api/ideas_workshop_ideas.feature [hunk 3]
```diff
@@ -869,13 +864,12 @@ Feature:
 
   Scenario: As a logged-in user I can publish my idea in the status DRAFT
```
Comment (illusionOfParadise) [inline features/api/ideas_workshop_ideas.feature]: Please, change also this scenario description.

Comment (ottaviano) [reply features/api/ideas_workshop_ideas.feature]: done tnx

File: features/api/ideas_workshop_ideas.feature (hunk 2 above)
Comment (ottaviano) [reply features/api/ideas_workshop_ideas.feature]: done tnx

File: features/api/ideas_workshop_ideas.feature (hunk 1 above)
Comment (ottaviano) [reply features/api/ideas_workshop_ideas.feature]: done

**Pull request merged**